src/utils.jl

```diff
@@ -70,10 +83,10 @@ function imagepreproc(trainimg, soft, geoconfig)
     replace!(AUX, NaN => 0.)
     replace!(AUXTI, NaN => 0.)
 
-    AUX, AUXTI
+    AUX, array_kernel(AUXTI)
```

Comment:
Why AUX is not being moved as well? Can we move all arrays at the very end of the function instead? So keep the code untouched and only at the end before returning you can call the array_kernel in everything.

Reply:
The AUX is not directly used in the `imfilter`, instead, only a tile is used as kernel. So its not worth the conversion.

Reply:
Makes sense. I think it is worth calling `array_kernel` at the end of the function anyways to make the code easier to follow and modify in the future.

Reply:
At the end you could simply return something like:

```julia
# move arrays to GPU if necessary
TIₖ = array_kernel(TI)
SOFTₖ = zip(first.(SOFT), array_kernel.(last.(SOFT)))

TIₖ, SOFTₖ